### DISCUSSION TITLE
Allow to disable the focus ring globally

### DISCUSSION THREAD

**Discussion by lcswillems** [chakra-ui/chakra-ui#9378] [Ideas]
### Description

To my knowledge there is no way to disable the focus ring globally.

### Problem Statement/Justification

I don't want the focus ring at any place.

### Proposed Solution or API

Adding a config to disable the focus ring globally.

**Comment by YagoBiermann** [comment]
Hello there, I was trying the same thing and found a solution(or workaround).

in your custom theme, add the property `focusRingStyle: "hidden"` to `globalCss` like so:

```
import { createSystem, defineConfig, defaultConfig } from "@chakra-ui/react";

const customConfig = defineConfig({
  globalCss: {
    "html, body": {
      focusRingStyle: "hidden"
    },
  },
});

export const system = createSystem(defaultConfig, customConfig);

```
now the focus ring should be disabled globally.

**Comment by lcswillems** [comment]
Thank you for sharing! The only thing is that the focus ring still appears on inputs for example if I'm not wrong?

**Comment by YagoBiermann** [comment]
You're welcome! I was testing it further and I find out that this workaround only works for Chakra components.

In order to work with all kinds of components you have to add both `focusRingStyle` and `focusRing` properties within the `universal selector`

```
const customConfig = defineConfig({
  globalCss: {
    "*": {
      focusRingStyle: "none",
      focusRing: "none",
    },
  },
});
```

Hope it helps.

**Comment by lcswillems** [comment]
It is even better. Thank you!

However, on my side, if I do this, it seems that I still have the same focus right on the inputs. But it seems that this focus ring alters directly the border itself. Do you what I'm referring too?

**Comment by YagoBiermann** [comment]
what kind of input are you using? Can you share the code and a screenshot of what you're seeing?

**Comment by lcswillems** [comment]
I'm using Chakra-UI one:

```
<Input />
```

There is a border white:

![image](https://github.com/user-attachments/assets/2ba6762f-575d-45ea-9167-ce21acc4e9dc)

![image](https://github.com/user-attachments/assets/754f7d53-53d7-4742-8d09-8bcd8e562277)

In my case, I've done `"*": { "focusRingWidth": 0 }`. I've also tried yours. It correctly removes the focus ring everywhere including on inputs, but on inputs it also changes the border color (cf "border-color" in the 2nd screenshot).

**Comment by segunadebayo** [comment]
You can set the `focusRingColor` globally or for a section of the image by assigning `--focus-ring-color` variable

```tsx
import { Button, Center } from '@chakra-ui/react';

function App() {
  return (
    <Center
      minH="dvh"
      css={{ '& *': { '--focus-ring-color': 'transparent !important' } }}
    >
      <Button variant="outline" size="sm">
        Hover me
      </Button>
    </Center>
  );
}

export default App;
```

**Reply by lcswillems** [reply]
@segunadebayo I don't think it works because it then removes the color of the border of the input?

**Reply by bdcorps** [reply]
> @segunadebayo I don't think it works because it then removes the color of the border of the input?

I'm seeing this as well. Any fix you found?

**Comment by segunadebayo** [comment]
Duplicate of https://github.com/chakra-ui/chakra-ui/discussions/9251

Set `focusVisibleRing: "none"` on the recipe, or `focusVisibleRing="none"` on the instance. https://chakra-ui.com/docs/styling/focus-ring
